src/jit/MemoryInl.h

```diff
@@ -215,11 +246,83 @@ void MemAddress::load(sljit_compiler* compiler)
 #endif /* HAS_SIMD */
 }
 
+#if defined(ENABLE_EXTENDED_FEATURES) && (defined SLJIT_32BIT_ARCHITECTURE && SLJIT_32BIT_ARCHITECTURE)
+static void atomicRmwLoad64(int64_t* shared_p, uint64_t* result)
+{
+    std::atomic<uint64_t>* shared = reinterpret_cast<std::atomic<uint64_t>*>(shared_p);
+    *result = shared->load(std::memory_order_relaxed);
+}
+
+static void atomicRmwStore64(uint64_t* shared_p, int64_t* value)
+{
+    std::atomic<uint64_t>* shared = reinterpret_cast<std::atomic<uint64_t>*>(shared_p);
+    shared->store(*value);
+}
+
+static void emitAtomicLoadStore64(sljit_compiler* compiler, Instruction* instr)
+{
+    uint32_t options = MemAddress::CheckNaturalAlignment | MemAddress::DontUseOffsetReg;
+    uint32_t size = 8;
+    sljit_u32 offset;
+
+    Operand* operands = instr->operands();
+    MemAddress addr(options, instr->requiredReg(0), instr->requiredReg(1), 0);
+
+    if (instr->opcode() == ByteCode::I64AtomicLoadOpcode) {
+        MemoryLoad* loadOperation = reinterpret_cast<MemoryLoad*>(instr->byteCode());
+        offset = loadOperation->offset();
+    } else if (instr->opcode() == ByteCode::I64AtomicStoreOpcode) {
+        MemoryStore* storeOperation = reinterpret_cast<MemoryStore*>(instr->byteCode());
+        offset = storeOperation->offset();
+    } else {
+        ASSERT_NOT_REACHED();
```

Comment:
these are too long for me. Just use `} else {` and add `ASSERT(instr->opcode() == ByteCode::I64AtomicStoreOpcode);`